symengine/lib/symengine/pywrapper.h

```diff
@@ -0,0 +1,117 @@
+#ifndef SYMENGINE_PYWRAPPER_H
+#define SYMENGINE_PYWRAPPER_H
+
+#include <Python.h>
+#include <symengine/number.h>
+#include <symengine/constants.h>
+#include <symengine/functions.h>
+
+namespace SymEngine {
+
+class PyModule : public EnableRCPFromThis<PyModule> {
+public:
+    PyObject* (*to_py_)(const RCP<const Basic>);
+    RCP<const Basic> (*from_py_)(PyObject*);
+    RCP<const Number> (*eval_)(PyObject*, long);
+    RCP<const Basic> (*diff_)(PyObject*, RCP<const Basic>);
+    PyObject *one, *zero, *minus_one;
+public:
+    PyModule(PyObject* (*)(const RCP<const Basic> x), RCP<const Basic> (*)(PyObject*),
+             RCP<const Number> (*)(PyObject*, long), RCP<const Basic> (*)(PyObject*, RCP<const Basic>));
+    ~PyModule();
+    PyObject* get_zero() const { return zero; }
+    PyObject* get_one() const { return one; }
+    PyObject* get_minus_one() const { return minus_one; }
+};
```

Comment:
Add documentation for `PyModule` about what it does and how it is meant to be used. If I understand it correctly, the higher level code must provide Python functions `from_py_`, `eval_` and `diff_`, that are stored in this `PyModule`. They are then called from C++.